### DISCUSSION TITLE
checkout-forms - brak stawki vat dla zamówienia

### DISCUSSION THREAD

**Discussion by Morph21** [allegro/allegro-api#8091] [Suggestions - resolved]
Występuje problem: 
 - [x] z zasobem REST API / with a REST API resource
 - [ ] ze środowiskiem produkcyjnym / with the production environment
 - [ ] ze środowiskiem testowym / with the test environment
 - [ ] z logowaniem lub autoryzacją / with login or authorization
 - [ ] na portalu https://developer.allegro.pl/ / at https://developer.allegro.pl/en/
 - [ ] inny / other

:beetle: **Opis / Description**

Aktualnie żeby wystawić fakturę zasób checkout-forms zwraca wszystkie potrzebne informacje, za wyjątkiem jednym - stawką vat.
Skutkuje to bezsensowną sytuacją, w której zamiast pobrać wszystkie potrzebne dane wywołując jeden zasób należy wywołać jeden zasób a później 20 razy drugi zasób (zakładając że /order/checkout-forms zwróci 20 wyników)

drugim zasobem w tym przypadku jest /sale/product-offers

Nie rozumiem dlaczego nie ma w zasobie checkout-forms informacji o stawce vat skoro to skutkuje tym że my musimy dla każdego klienta robić ogromną ilość zapytań do allegro, allegro niepotrzebnie dostaje taką dużą ilość zapytań, a klienci nie potrzebnie czekają 20 razy dłużej na informację zwrotną.

Dla przykładu jeżeli mam klienta który ma 4000 zamówień i chce pobrać je wszystkie to wywołamy 200 razy zasób /orders/checkout-forms i 4000 zasób /sale/product-offers zamiast po prostu wywołać 200 razy checkout-forms.

**Proszę o rozważenie dodania informacji o stawce vat dla endpointu /order/checkout-forms**
Ułatwiło by to pracę wielu waszym klientom, naszym klientom i klientom każdej firmy która korzysta z tego api i wystawia jakąkolwiek fakturę/paragon itd

**Comment by Lukasz-Zurek** [comment]
@Morph21 Dziękuję za Twoją sugestię. Przeniosłem wątek do zakładki "Discussions", w której zbieramy różnego rodzaju pomysły na usprawnienie Allegro API. Jednocześnie głos na ten pomysł oddać mogą również inni użytkownicy.

Co prawda podobna sugestia była już jakiś czas temu w [tym](https://github.com/allegro/allegro-api/discussions/6958) wątku. W maju otrzymaliśmy informację, że nie mamy takich planów co do zwracania stawki VAT w zamówieniach w API. Niemniej jednak od tamtego czasu minęło już około pół roku, dlatego wątek ten będzie weryfikowany. Bardzo Ci dziękuję za konkretne uzasadnienie.

**Reply by Morph21** [reply]
Mam nadzieję że może tym razem wynik będzie inny. Przesyłanie stawki vat w liście zamówień powinno być standardem w branży ecommerce, nie feature-m. Każdy inny serwis ecommerce  z którym mamy integracje podaje takie dane

**Comment by yao657355700** [comment]
It is indeed necessary to increase the nodes of tax rates

**Comment by SimasBir** [comment]
+1
Additionally, 
recently came across a problem when trying to get earlier orders info (2-3 month old). Trying to get tax rate for the line items from the /sale/product-offers endpoint, the offers are missing:

```
"errors": [
        {
            "code": "OfferNotFoundException",
            "message": "Not Found",
            "details": null,
            "path": "",
            "userMessage": "Offer 14707549116 does not exist."
        }
    ]
```
Fairly certain they existed before.

Is there any other way to get tax rate in this case?

**Reply by Lukasz-Zurek** [reply]
This offer has been already archived (within 60 days of expiring an offer, information is archived), so you can't retrieve it.

**Reply by SimasBir** [reply]
Thanks for clarifying that up.

The main question is still - is there no way to get the lineitem tax rate in that case?

**Reply by Lukasz-Zurek** [reply]
For the offer which has been already archived - it won't be possible.

**Comment by arekpietrzyk** [comment]
+1 dla zwracania zdefiniowanej przy ofercie stawki VAT. Obecnie mamy to zdefiniować w ofercie, serwis przelicza kwotę netto, a do post-processingu zamówień trzeba wróżyć stawkę z zewnętrznych źródeł.

**Comment by AureliuszBrussy** [comment]
Właśnie wdrożyliśmy stawkę VAT w zamówieniach: https://github.com/allegro/allegro-api/issues/8747

**Comment by Morph21** [comment]
Pięknie 🎉 
Teraz będzie api śmigać :D
